Release/release 0.1.13

Senkrad68 opened the pull request.

eviltrout commented on assets/javascripts/discourse/components/admin-disagree-flag-dropdown.js.es6:
```diff
@@ -0,0 +1,100 @@
+
+    disagree_other(reason) {
+      let flaggedPost = this.get("post");
+      let otherReason = prompt("Please enter the reason:", "correct based on the context");
```
These strings should use i18n.

Senkrad68 replied:
Fixed

eviltrout commented on assets/javascripts/discourse/connectors/flagged-post-below-controls/sift-flag-queue.js.es6:
```diff
@@ -0,0 +1,38 @@
+
+export default {
+
+    internalReportingDone: (reason) => {
```
This is a bit odd. Using the fat arrow syntax means that `this` doesn't point to your singleton anymore. Additionally this doesn't seem to be used?

Senkrad68 replied:
That was the last attempt I made to get a common function that could be called, but could not get it to work.  Removed

eviltrout commented on assets/javascripts/discourse/templates/admin/plugins-sift.hbs:
```diff
@@ -40,33 +40,38 @@
+                        {{/if}}
+                    </div>
+                {{~else~}}
+                    <span>No Sift Info available</span>
```
This is not translated either?

Senkrad68 replied:
fixed

eviltrout commented on assets/javascripts/helpers/ifCond.js.es6:
```diff
@@ -0,0 +1,17 @@
+import { registerHelper } from 'discourse-common/lib/helpers';
+
+
+registerHelper('ifCond', function(params) {
```
Since you're only using this helper in one place, it might make more sense to use a more specific one to your purpose. Additionally, you could create a computed property in a component that does the check (that's usually how we do it in discourse.)

Senkrad68 replied:
Re-jiggered and renamed the helper to be more specific.

eviltrout commented on assets/stylesheets/sift_classification.scss:
```diff
@@ -15,77 +15,175 @@ $topic_list: 'General', 'Bullying', 'Fighting', 'PII', 'Sexting', 'Vulgar', 'Dru
+  padding: 3px 6px;
+
+  color: white;
+  background-color: #999999;
```
I suspect this is not theme friendly in Discourse. Some forums are dark, some are light. It would be good to use discourse's variables and helpers to set your colors.

Senkrad68 replied:
Fixed

eviltrout commented on jobs/report_post.rb:
```diff
@@ -0,0 +1,26 @@
+      post = Post.where(id: args[:post_id], user_deleted: false).first
+      return unless post.present?
+
+      Sift::Client.with_client() do |client|
```
```suggestion
      Sift::Client.with_client do |client|
```

Senkrad68 replied:
Fixed

eviltrout commented on lib/sift.rb:
```diff
@@ -70,21 +70,22 @@ def topic_string
 
-        def self.with_client(base_url:, api_key:, api_url:, end_point:)
-          client = self.new(base_url: base_url, api_key: api_key, api_url: api_url,  end_point: end_point)
+        def self.with_client()
```
```suggestion
        def self.with_client
```

Senkrad68 replied:
fixed

Senkrad68 commented:
Thanks for the review Robin.  The client these changes are for are not going to be moving the reviewable version of the code for a while but needed these changes right away so we had to fix this version.  I think I have addressed all your comments.
